### DISCUSSION TITLE
Event before asset is uploaded

### DISCUSSION THREAD

**Discussion by BrandonOldenhof** [statamic/cms#8018] [Q&A] [answered]
Hello!

I'm looking for a good way to optimize assets using imagick or gd; I want to convert jpg and png images to webp to save on disk usage and page loading speed.
Ideally I would listen for a Statamic event after upload has finished but before the various presets are generated.

Is there such an event, or something similar that will allow me to do the same thing?

**Comment by ryanmitchell** [comment, marked as answer]
You can process source file uploads: https://statamic.dev/image-manipulation#process-source-images - maybe this is close enough to what you are looking for?

**Reply by ryanmitchell** [reply]
Failing that, there is also the AssetUploaded event which you could listen for.
